File: src/main/resources/templates/angular-app/src/interfaces/cursoProfesorDTO.ts
export class CursoProfesorDTO {
    curso!:CursoDTO;
    profesores!: ProfesorDTO[];
  }
  
  export class ProfesorDTO {
    id!:number;
    nombresprof!: string;
    apePatprof!: string;
    apeMatProf!: string;
    espeprof!: string;
    inicio!:Date;
    fin!:Date;
    isChecked!:boolean;
  }
  export class CursoDTO {
    idcurso!: number;
    nomcurso!: string;
  }
  

File: src/main/java/com/cibertec/dto/CursoProfesorDTO.java
package com.cibertec.dto;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import com.cibertec.model.Profesor;

import lombok.Data;

@Data
public class CursoProfesorDTO {
	
	private CursoDTO curso=new CursoDTO();
	
	private List<ProfesorDTO>profesores=new ArrayList<>(); 
	public CursoProfesorDTO(Long id, String nombre,LocalDateTime inicio,LocalDateTime fin,Profesor p) {
		curso.setIdcurso(id);
		curso.setNomcurso(nombre);
		profesores.add(new ProfesorDTO(p.getIdprof(),p.getNombresprof(),p.getApePatProf(),p.getApeMatProf(),p.getEspeprof(),inicio,fin));
	}
	public CursoProfesorDTO(Long id, String nombre) {
		curso.setIdcurso(id);
		curso.setNomcurso(nombre);
	}

}

